docs/development-process.md

```diff
@@ -0,0 +1,32 @@
+# imgui-rs development process
+
+As summarised in [issue #665](https://github.com/imgui-rs/imgui-rs/issues/665)
+
+In summary:
+
+1. There is a `main` branch
+2. For each semver compatible release there is a stable branch (e.g `0.9-stable`)
+3. Each patch release becomes a tagged commit on this stable branch (e.g `v0.9.5` would come from a tagged commit on the `0.9-stable` branch)
+
+## General process
+
+Day to day development
+
+1. Work on `main` branch
+2. General PR's are submitted against the `main` branch
+
+When it is time to make a new release, we create a `x.y-stable` branch (e.g `0.9-stable`) from `main`
+
+1. Ensure `CHANGELOG` is up to date
+1. Bump `version` in the various `Cargo.toml`
+2. A stable branch is created, e.g `git switch -c 0.9-stable` and pushed to Github
+
+All further PR's are still done to `main`
+
+1. If they are applicable to previous release (e.g bugfixes or non-breaking changes), they are cherry-picked back to the applicable `stable` branch(es)
```

Comment:
How do you want this to happen? Should this be discussed/described in a PR beforehand? Then, should a maintainer perform the `cherry-pick` as soon as a PR is merged so that it is not forgotten about?

Or should this be a more dynamic process, where a maintainer (in an issue/discussion) at some point decides pick/backport a list of merged PRs and publish a patch release?

It depends on the kind and volume of changes/contributions a repository receives, keeping up with `cherry-pick`s might turn out to be a lot of work but also prevents the stable branch from lagging behind and turning into a gigantic pile of work when it comes to making a patch release.

---

Perhaps a more rigorous maintenance plan needs to be put in place (how often to allow breaking releases, how long to support how many older stable versions, whether to at least publish a patch release with accumulated fixes/additions right before doing the next breaking release, etc.), but that can also form over time pending how this release process pans out (again: depends on how many breaking changes vs purely additions/bugfixes this repo receives).

Reply:
These are good points, but I suspect we should come back to this later in the interests of progress

For now I think the PR's would just go to the development branch, and if required be backported to stable by whoever merges them (e.g click merge, decide if it needs backported)

Once we are back to a less "abandonded-looking" development cycle, we should definitely discuss this further!